modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/ImportAndExportObjects.testcase

```diff
@@ -0,0 +1,267 @@
+
+			ImportExport.exportFile(
+				entityType = "C_Subject (v1_0 - Liferay Object REST)",
+				exportFields = "Attribute Code",
+				exportFileFormat = "JSON");
+		}
+
+		task ("Then zip file contains related file type export successfully") {
+			ImportExport.unzipDownloadedExportFile();
+
+			BatchEngine.assertExportedFileContainsCorrectObject(
+				expectedValue = "Math",
+				fileName = "export.json",
+				jsonObject = "$..[0]name");
+		}
+	}
+
+	@priority = 4
+	test ExportCustomObjectEntryWithOneToManyRelationship {
+		property portal.acceptance = "true";
+
+		task ("Given two custom objects published") {
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "university",
+				en_US_plural_label = "universities",
+				name = "University",
+				requiredStringFieldName = "name");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "student",
+				en_US_plural_label = "students",
+				name = "Student",
+				requiredStringFieldName = "name");
+		}
+
+		task ("And Given oneToMany releationship has been set between them") {
+			ObjectDefinitionAPI.createRelationshipWithObjectDefinitionNames(
+				childObjectName = "Student",
+				deletionType = "cascade",
+				name = "universityStudents",
+				parentObjectName = "University",
+				type = "oneToMany");
+		}
+
+		task ("And Given one entry related to antoher one") {
+			var universityEntryId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "universities",
+				name = "Oxford");
+
+			ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "students",
+				name = "Pearl");
+
+			ObjectDefinitionAPI.relateObjectEntriesByName(
+				child_plural_label = "students",
+				childEntryName = "Pearl",
+				parent_plural_label = "universities",
+				parentEntryName = "Oxford",
+				relationshipName = "universityStudents");
+		}
+
+		task ("When user export custom object as entity type") {
+			ImportExport.openImportExportAdmin();
+
+			ImportExport.exportFile(
+				entityType = "C_Student (v1_0 - Liferay Object REST)",
+				exportFields = "Attribute Code",
+				exportFileFormat = "JSON");
+		}
+
+		task ("Then zip file contains related file type export successfully") {
+			ImportExport.unzipDownloadedExportFile();
+
+			BatchEngine.assertExportedFileContainsCorrectObject(
+				expectedValue = "Pearl",
+				fileName = "export.json",
+				jsonObject = "$..[0]name");
+		}
+	}
+
+	@priority = 4
+	test ExportCustomObjectWithAllFieldTypesMapped {
+		property portal.acceptance = "true";
+
+		task ("Given a custom object with all kinds of field type is publushed") {
+			ObjectDefinitionAPI.createObjectDefinitionWithAllKindsOfFieldTypes(
+				en_US_label = "student",
+				en_US_plural_label = "students",
+				objectDefinitionName = "Student",
+				requiredStringFieldName = "name");
+		}
+
+		task ("And Given object entry is created") {
+			var studentEntryId = ObjectDefinitionAPI.createObjectEntryWithName(
```

Comment:
Just wondering instead of ObjectFieldAPI.updateAllFieldsOfEntryById(), can we create a macro that something customObjectAPI.createEntryOfAllFieldsById?

Reply:
it's not a post method it's a patch method